src/material-experimental/mdc-slider/slider.ts

```diff
@@ -20,18 +20,146 @@ import {
   ChangeDetectorRef,
   Component,
   ContentChildren,
+  Directive,
   ElementRef,
+  EventEmitter,
   Inject,
   Input,
   OnDestroy,
+  Output,
   QueryList,
   ViewChild,
   ViewChildren,
   ViewEncapsulation,
 } from '@angular/core';
-import {MDCSliderFoundation, Thumb, TickMark} from '@material/slider';
-import {SliderAdapter} from './slider-adapter';
-import {MatSliderThumb} from './slider-thumb';
+import {SpecificEventListener, EventType} from '@material/base';
+import {MDCSliderAdapter, MDCSliderFoundation, Thumb, TickMark} from '@material/slider';
+
+/**
+ * Represents a drag event emitted by the MatSlider component.
+ */
+export interface MatSliderDragEvent {
+  /** The MatSliderThumb that was interacted with. */
+  source: MatSliderThumb;
+
+  /** The MatSlider that was interacted with. */
+  parent: MatSlider;
+
+  /** The current value of the slider. */
+  value: number;
+
+  /** The thumb that was interacted with. */
+  thumb: Thumb;
```

Comment:
Can we remove this from the event? You can just do `source._thumb` to get it. I don't think we should expose this type in a public facing API - its part of MDC and therefore more of an implementation detail